docs/pages/api-docs/data-grid/grid-api.md

```diff
@@ -10,92 +10,92 @@ import { GridApi } from '@mui/x-data-grid-pro';
+| <span class="prop-name">showFilterPanel</span>         | <span class="prop-type">(targetColumnField?: string \| undefined) =&gt; void</span>                                                                                                                                                                                                                                                                                                     | Shows the filter panel. If `targetColumnField` is given, a filter for this field is also added.                                                                |
+| <span class="prop-name">showPreferences</span>         | <span class="prop-type">(newValue: GridPreferencePanelsValue) =&gt; void</span>                                                                                                                                                                                                                                                                                                         | Displays the preferences panel. The `newValue` argument controls the content of the panel.                                                                     |
+| <span class="prop-name">sortColumn</span>              | <span class="prop-type">(column: GridColDef, direction?: GridSortDirection, allowMultipleSorting?: boolean \| undefined) =&gt; void</span>                                                                                                                                                                                                                                              | Sorts a column.                                                                                                                                                |
+| <span class="prop-name">state</span>                   | <span class="prop-type">GridState</span>                                                                                                                                                                                                                                                                                                                                                | Property that contains the whole state of the grid.                                                                                                            |
+| <span class="prop-name">subscribeEvent</span>          | <span class="prop-type">&lt;E extends "resize" \| "debouncedResize" \| "viewportInnerSizeChange" \| "componentError" \| "unmount" \| "cellModeChange" \| "cellClick" \| "cellDoubleClick" \| "cellMouseDown" \| "cellMouseUp" \| ... 57 more ... \| "virtualScrollerContentSizeChange"&gt;(event: E, handler: GridEventListener&lt;...&gt;, options?: EventListenerOptions \|...</span> | Registers a handler for an event.                                                                                                                              |
```

Comment:
"... 57 more ..."

With TypeDoc it was not very good because there was too many interfaces, but with TS not showing all options is also bad. Is it possible to not explode the keys of `GridEventsStr`? Keep only "E extends GridEventsStr"?

Reply:
This is bad I agree
With TypeDoc it was not possible
I'll check if with pure TS it is. And if not I can add a custom code for that property.

Reply:
I got the option to remove the truncation at least